experimental/packages/opentelemetry-sdk-metrics-base-wip/src/Meter.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as metrics from '@opentelemetry/api-metrics';
+import { InstrumentationLibrary } from '@opentelemetry/core';
+import { Measurement } from './Measurement';
+import { MeterProvider } from './MeterProvider';
+import { Histogram, Metric } from './Metric';
+
+export class Meter implements metrics.Meter {
+    // instrumentation library required by spec to be on meter
+    // spec requires provider config changes to apply to previously created meters, achieved by holding a reference to the provider
+    constructor(private _provider: MeterProvider, private _instrumentationLibrary: InstrumentationLibrary, private _schemaUrl?: string) { }
+
+    /** this exists just to prevent ts errors from unused variables and may be removed */
+    getSchemaUrl(): string | undefined {
+        return this._schemaUrl;
+    }
+
+    /** this exists just to prevent ts errors from unused variables and may be removed */
+    getInstrumentationLibrary(): InstrumentationLibrary {
+        return this._instrumentationLibrary;
+    }
+
+    createHistogram(_name: string, _options?: metrics.MetricOptions): Histogram {
+        throw new Error('Method not implemented.');
+    }
+    createCounter(_name: string, _options?: metrics.MetricOptions): metrics.Counter {
+        throw new Error('Method not implemented.');
+    }
+    createUpDownCounter(_name: string, _options?: metrics.MetricOptions): metrics.UpDownCounter {
+        throw new Error('Method not implemented.');
+    }
+    createObservableGauge(_name: string, _options?: metrics.MetricOptions, _callback?: (observableResult: metrics.ObservableResult) => void): metrics.ObservableBase {
+        throw new Error('Method not implemented.');
+    }
+    createObservableCounter(_name: string, _options?: metrics.MetricOptions, _callback?: (observableResult: metrics.ObservableResult) => void): metrics.ObservableBase {
+        throw new Error('Method not implemented.');
+    }
+    createObservableUpDownCounter(_name: string, _options?: metrics.MetricOptions, _callback?: (observableResult: metrics.ObservableResult) => void): metrics.ObservableBase {
+        throw new Error('Method not implemented.');
+    }
+
+    public aggregate(metric: Metric, measurement: Measurement) {
```

Comment:
Do we want this to be public, since users aren't meant to call it. Alternative would be to pass a private method to each instrument as a callback

Reply:
I wish node/ts had a package level privacy feature. Let me think about this.

Reply:
In the Trace API we do something similar to what is done here. The SDK has many public methods which are meant for internal use. As long as the user uses the API and API types, they aren't an issue. For now, I'm going to leave this for simplicity, but I am open to a PR which changes it.